encryption-client/src/main/java/org/wso2/mi/tooling/security/output/OutputManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.mi.tooling.security.output;
+
+import org.wso2.mi.tooling.security.Constants;
+import org.wso2.mi.tooling.security.exception.EncryptionToolException;
+import org.wso2.mi.tooling.security.utils.IOUtils;
+
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This class manages encrypted vale output.
+ */
+public class OutputManager {
+
+    private String outputType;
+    private Map<String, String> secrets;
+
+    public OutputManager(String outputType, Map<String, String> encodedValues) {
+
+        this.outputType = outputType;
+        secrets = encodedValues;
+    }
+
+    /**
+     * Sets the output based on the output type.
+     */
+    public void setOutput() {
+
+        if (Constants.PROP_CONSOLE_OUTPUT.equalsIgnoreCase(outputType)) {
+            doConsoleOutput();
+        } else if (Constants.PROP_FILE_OUTPUT.equalsIgnoreCase(outputType)) {
+            doFileOutput();
+        } else if (Constants.PROP_K8_OUTPUT.equalsIgnoreCase(outputType)) {
+            doK8Output();
+        } else {
+            throw new EncryptionToolException("Secret output type is not supported.");
```

Comment:
Let's print the allowed types and the type that was passed.

Reply:
fixed